pkg/querier/frontend/frontend.go

```diff
@@ -38,8 +38,9 @@ var (
 		Help:      "Number of queries in the queue.",
 	})
 
-	errTooManyRequest = httpgrpc.Errorf(http.StatusTooManyRequests, "too many outstanding requests")
-	errCanceled       = httpgrpc.Errorf(http.StatusInternalServerError, "context cancelled")
+	errTooManyRequest   = httpgrpc.Errorf(http.StatusTooManyRequests, "too many outstanding requests")
+	errCanceled         = httpgrpc.Errorf(499, context.Canceled.Error())
```

Comment:
499 seems appropriate here (https://httpstatuses.com/499)

Nit: I would create a constant for it, `StatusClientClosedRequest`, and use said constant in the test.

Reply:
Will do!